File: plugins/test-fixture/src/server.test.js
import assert from "node:assert/strict";
import test from "node:test";

import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { InMemoryTransport } from "@modelcontextprotocol/sdk/inMemory.js";

import { createFixtureServer, FIXTURE_TOOL_NAME } from "./server.js";

test("the fixture is usable by an ordinary MCP client", async () => {
  const [clientTransport, serverTransport] =
    InMemoryTransport.createLinkedPair();
  const client = new Client({ name: "fixture-test-client", version: "0.1.0" });
  const server = createFixtureServer();

  try {
    await server.connect(serverTransport);
    await client.connect(clientTransport);
    const tools = await client.listTools();
    const fixture = tools.tools.find((tool) => tool.name === FIXTURE_TOOL_NAME);
    assert.equal(fixture?.annotations?.readOnlyHint, true);

    const result = await client.callTool({
      name: FIXTURE_TOOL_NAME,
      arguments: {},
    });
    assert.deepEqual(result.structuredContent, {
      source: "agent-hub-foundation-fixture",
      status: "available",
    });
    assert.equal(result.content[0]?.type, "text");
  } finally {
    await clientTransport.close();
  }
});

File: plugins/test-fixture/src/agent_hub_foundation_fixture/_server_test.py
import pytest

fastmcp = pytest.importorskip("fastmcp")

from fastmcp.client import Client

from agent_hub_foundation_fixture._server import (
    FIXTURE_APP_RESOURCE_URI,
    FIXTURE_TOOL_NAME,
    create_fixture_server,
)


@pytest.mark.asyncio
async def test_the_fixture_is_usable_by_an_ordinary_mcp_client() -> None:
    async with Client(create_fixture_server(), timeout=1, init_timeout=1) as client:
        tools = await client.list_tools()
        assert [tool.name for tool in tools] == [FIXTURE_TOOL_NAME]
        fixture_tool = tools[0]
        assert fixture_tool.annotations is not None
        assert fixture_tool.annotations.read_only_hint is True

        result = await client.call_tool(FIXTURE_TOOL_NAME, {})
        assert result.data == {
            "source": "agent-hub-foundation-fixture",
            "status": "available",
        }

        resources = await client.list_resources()
        assert any(str(resource.uri) == FIXTURE_APP_RESOURCE_URI for resource in resources)
        contents = await client.read_resource(FIXTURE_APP_RESOURCE_URI)
        assert contents[0].mime_type == "text/html;profile=mcp-app"
        assert "ui/initialize" in contents[0].text
